pkg/config/autodiscover/autodiscover_targets.go

```diff
@@ -138,26 +146,32 @@ func GetNodesList() (nodes map[string]configsections.Node) {
 	return nodes
 }
 
-// FindTestDeploymentsByLabel uses the containers' namespace to get its parent deployment. Filters out non CNF test deployments,
+// FindTestPodSetsByLabel uses the containers' namespace to get its parent deployment. Filters out non CNF test deployments,
 // currently partner and fs_diff ones.
-func FindTestDeploymentsByLabel(targetLabels []configsections.Label, target *configsections.TestTarget) (deployments []configsections.Deployment) {
+func FindTestPodSetsByLabel(targetLabels []configsections.Label, target *configsections.TestTarget, resourceTypeDeployment string) (podsets []configsections.PodSet) {
 	for _, label := range targetLabels {
-		deploymentResourceList, err := GetTargetDeploymentsByLabel(label)
+		podsetResourceList, err := GetTargetPodSetsByLabel(label, resourceTypeDeployment)
+		Type := configsections.Deployment
+		if resourceTypeDeployment == string(configsections.StateFullSet) {
+			Type = configsections.StateFullSet
+		}
```

Comment:
resourceTypeDeployment is not changing inside the loop, so we could take these 4 lines and put them at the beginning, outside of the loop.